eng/docgeneration/script/docgen.ps1

```diff
@@ -0,0 +1,61 @@
+﻿# Extract Dlls from nuget package
+# Copy the items to the dll directory
+function ExtractDlls($PkgDirPath, $DllDirPath) {
+    Set-Location $PkgDirPath
+    $PkgList = Get-ChildItem -Path $PkgDirPath
+
+    foreach ($PkgName in $PkgList) {
+        if (!$PkgName.Name.EndsWith('snupkg')) {
+            Rename-Item -Path $PkgName.Name -NewName "$($PkgName.BaseName).zip"
+            Expand-Archive -Path "$($PkgName.BaseName).zip"
+            Remove-Item -Path "$($PkgName.BaseName).zip" -Force
+
+            $DllPath = Join-Path -Path $PkgDirPath -ChildPath "$($PkgName.BaseName)/lib/netstandard2.0/*"
+
+            if (Test-Path -Path $DllPath) {
+                Copy-Item $DllPath -Destination $DllDirPath
+            }
+            else {
+                Write-Host "$($DllPath) does not exist"
+            }
+        }
+        else {
+            Remove-Item -Path $PkgName.FullName -Force
+        }
+    }
+}
+
+# Create Directories used in the process
+function CreateDocGenFolders() {
+    New-Item -Type dir dll-docs/my-api
+    New-Item -Type dir dll-docs/dependencies/my-api
+    New-Item -Type dir dll-xml-output
+    New-Item -Type dir dll-yaml-output
+    New-Item -Type dir docfx-output
+}
+
+# Include dependent Dlls and XML files
+function IncludeDependencies($SourcesDirectory, $BinariesDirectory) {
+    Copy-Item -Path "$($SourcesDirectory)/tools/LocalNugetFeed/PopImport.1.0.0.nupkg" -Destination $BinariesDirectory
+    Invoke-WebRequest -Uri "https://www.nuget.org/api/v2/package/Microsoft.Bcl.AsyncInterfaces/1.0.0-preview6.19303.8"`
+    -OutFile "Microsoft.Bcl.AsyncInterfaces.zip" | Wait-Process; Expand-Archive -Path "Microsoft.Bcl.AsyncInterfaces.zip" -DestinationPath "$($BinariesDirectory)/microsoft.bcl.asyncinterfaces"
+    Rename-Item -Path "$($BinariesDirectory)/PopImport.1.0.0.nupkg" -NewName "PopImport.zip"
```

Comment:
We should try to avoid manually downloading and extracting these packages. Instead we should have a dummy project that we restore and if necessary copy the output after restore.

Reply:
Updated using dummy project